### DISCUSSION TITLE
MySQL support for returning auto generated ID on INSERT?

### DISCUSSION THREAD

**Discussion by mportugal** [sqlc-dev/sqlc#1032] [Feature Requests & Ideas] [answered]
Is it possible to support to retrieve the last ID upon INSERT? I know MySQL supports SELECT LAST_INSERT_ID() but I was wondering the INSERT generated by sqlc can at least return the auto genereted ID

**Comment by wensenz** [comment]
你可以在插入时指定ID， 比如UUID 或者 其他形式的自增ID。

**Comment by Jille** [comment, marked as answer]
For Postgres you can add `RETURN id` to your query and use `:one`. For MySQL, use `:execresult` which makes your query return a `sql.Result`, which has a `LastInsertId() (int64, error)` method.
